aws_change_observer/aws_change_observer_stack.py

```diff
@@ -107,16 +91,46 @@ def __init__(self, scope: Construct, construct_id: str, is_prod: bool = False, *
             handler="get_marker_request_lambda_function.lambda_handler",
             code=aws_lambda.Code.from_asset(GET_MARKER_REQUEST_LAMBDA_CODE_PATH),
             layers=[shared_classes_layer],
-            role=get_marker_role,
+            role=lambda_role,
             environment={
                 'TABLE_NAME': table.table_name,
             },
         )
 
+        # Lambda function for updating a marker
+        update_marker_request_lambda = aws_lambda.Function(
+            self, 'UpdateMarkerRequestFunction',
+            function_name='updateMarkerRequest',
+            runtime=aws_lambda.Runtime.PYTHON_3_8,
+            handler="update_marker_request_lambda_function.lambda_handler",
+            code=aws_lambda.Code.from_asset(UPDATE_MARKER_REQUEST_LAMBDA_CODE_PATH),
+            layers=[shared_classes_layer],
+            role=lambda_role,
+            environment={
+                'TABLE_NAME': table.table_name
+            },
+        )
+
+        # Lambda function for deleting a marker
+        delete_marker_request_lambda = aws_lambda.Function(
+            self, 'DeleteMarkerRequestFunction',
+            function_name='deleteMarkerRequest',
+            runtime=aws_lambda.Runtime.PYTHON_3_8,
+            handler="delete_marker_request_lambda_function.lambda_handler",
+            code=aws_lambda.Code.from_asset(DELETE_MARKER_REQUEST_LAMBDA_CODE_PATH),
+            layers=[shared_classes_layer],
+            role=lambda_role,
+            environment={
+                'TABLE_NAME': table.table_name
+            },
+        )
+
         # Grant access to the DynamoDB table
         table.grant_read_data(get_markers_request_lambda)
         table.grant_read_data(get_marker_request_lambda)
         table.grant_write_data(add_marker_request_lambda)
+        table.grant_write_data(update_marker_request_lambda)
```

Comment:
@PCBush you may also need to give **update_marker_request_lambda** read access because here in `update_marker()` there's also a read happening
https://github.com/miitchellstahl/aws-change-observer/blob/971ec9731be167741b9d22c7b660d922c107f5ce/layers/shared_classes_layer/python/data_service.py#L81

Reply:
Got it